dev_docs/key_concepts/persistable_state.mdx

```diff
@@ -0,0 +1,68 @@
+---
+id: kibDevDocsPersistableStateIntro
+slug: /kibana-dev-docs/persistable-state-intro
+title: Persistable State
+summary: Persitable state is a key concept to understand when building a Kibana plugin.
+date: 2021-02-02
+tags: ['kibana','dev', 'contributor', 'api docs']
+---
+
+ “Persistable state” is developer-defined state that supports being persisted by a plugin other than the one defining it. Persistable State needs to be serializable and the owner can/should provide utilities to migrate it, extract and inject references from/to it, as well as telemetry collection utilities.
+ 
+## Exposing state that can be persisted
+
+Any plugin that exposes state or configuration that another plugin might persist should implement `PersistableStateService`
+interface on their `setup` contract. This will allow plugins persisting the state to easily access migrations and other utilities.
+
+example: Data plugin allows you to generate filters. Those filters can be persisted by applications in their saved 
+objects or in the URL. In order to allow apps migrating the filters in case the structure changes in the future Data 
+plugin implements `PersistableStateService` on `data.query.filterManager`.
+
```

Comment:
So we can get this merged, I think we should highlight some of the risks with the current architecture. Maybe...


```suggestion
Any plugin that exposes state that another plugin can persist should implement `PersistableStateService`
interface on their `setup` contract. This will allow plugins persisting the state to access migrations and other utilities. 

There is currently no obvious way for a plugin to know which state is safe to persist. The developer must manually look for a matching `PersistableStateService`, or ad-hoc provided migration utilities (as is the case with Rule Type Parameters).

We already have situations where state is being persisted, even though no migration utilities are exposed. For example, filters from the data plugin are persisted in URLs and saved objects, but don't currently expose a `PersistableStateService`. This is something we will need to fix.

In the future, we hope to make it easier for producers of state to understand when they need to write a migration with changes, and also make it easier for consumers of such state, to understand whether it is safe to persist. 
```